src/app/(main)/mypage/_components/user-social-info/UserSocialInfoSection.tsx

```diff
@@ -3,112 +3,154 @@ import { useForm } from 'react-hook-form'
 
 import { zodResolver } from '@hookform/resolvers/zod'
 import { GitHubLogoIcon, InstagramLogoIcon } from '@radix-ui/react-icons'
+import { useMutation } from '@tanstack/react-query'
 
-import { Button, Form, Input } from '@/components/ui'
+import {
+  Button,
+  Form,
+  FormControl,
+  FormDescription,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+  Input,
+  toast,
+} from '@/components/ui'
+import { queryClient } from '@/lib/query-client'
 import {
   CreateMypageSocialInfo,
   CreateMypageSocialInfoSchema,
 } from '@/schema/mypage'
+import { putUpdateProfileApi, userQueries } from '@/service/api/mypage'
 
 interface UserSocialInfoSectionProps {
-  githubInfo: string
-  instagramInfo: string
-  profileIntro: string
+  githubInfo?: string
+  instagramInfo?: string
+  profileIntro?: string
+  userId: string
 }
 
 export const UserSocialInfoSection = ({
   githubInfo: initialGithubInfo,
   instagramInfo: initialInstagramInfo,
   profileIntro: initialProfileIntro,
+  userId,
 }: UserSocialInfoSectionProps) => {
-  const [isEditingIntro, setIsEditingIntro] = useState(false)
-  const [isEditingSocial, setIsEditingSocial] = useState(false)
+  const [isEditing, setIsEditing] = useState(false)
+
+  const { mutate: updateProfile, isPending } = useMutation({
+    mutationFn: putUpdateProfileApi,
+    onSuccess: (data) => onSuccess(data.message),
+  })
 
   const form = useForm<CreateMypageSocialInfo>({
     resolver: zodResolver(CreateMypageSocialInfoSchema),
     defaultValues: {
-      introduction: initialProfileIntro,
-      githubInfo: initialGithubInfo,
-      instagramInfo: initialInstagramInfo,
+      profileIntro: initialProfileIntro ?? undefined,
+      githubAccount: initialGithubInfo ?? undefined,
+      instaAccount: initialInstagramInfo ?? undefined,
     },
   })
 
-  const { register, handleSubmit, watch } = form
+  const onSuccess = (message?: string) => {
+    toast({
+      title: message,
+      duration: 2000,
+    })
+
+    queryClient.invalidateQueries({
+      queryKey: userQueries.profiles({ userId: userId }),
+    })
+  }
 
   const onSubmit = (data: CreateMypageSocialInfo) => {
-    console.log('제출된 데이터:', data)
-    setIsEditingIntro(false)
-    setIsEditingSocial(false)
+    updateProfile({
+      userId: userId,
+      profileData: {
+        profileIntro: data.profileIntro,
+        githubAccount: data.githubAccount,
+        instaAccount: data.instaAccount,
+      },
+    })
+    setIsEditing(false)
   }
 
-  const clickToEdit = (e: React.MouseEvent<HTMLButtonElement>) => {
+  const onClickEdit = (e: React.MouseEvent<HTMLButtonElement>) => {
     e.preventDefault()
-
-    if (isEditingIntro === false && isEditingSocial === false) {
-      setIsEditingIntro(true)
-      setIsEditingSocial(true)
-    } else {
-      setIsEditingIntro(false)
-      setIsEditingSocial(false)
-    }
+    setIsEditing(!isEditing)
```

Comment:
```suggestion
    setIsEditing(!isEditing)
    form.reset()
```

취소 버튼을 눌렀을 때, form input을 default 값으로 초기화하면 좋을 거 같아요

Reply:
수정했습니다.